lambdas/services/bulk_upload_report_service.py

```diff
@@ -158,16 +158,19 @@ def generate_success_report(self, ods_reports: list[OdsReport]):
                     [str(patient[0]), str(report.uploader_ods_code), str(patient[1])]
                 )
 
-        self.write_additional_report_items_to_csv(
-            file_name=file_name, headers=headers, rows_to_write=data_rows
-        )
+        if data_rows:
+            self.write_additional_report_items_to_csv(
+                file_name=file_name, headers=headers, rows_to_write=data_rows
+            )
 
-        logger.info("Uploading daily success report file to S3")
-        self.s3_service.upload_file(
-            s3_bucket_name=self.reports_bucket,
-            file_key=f"{self.s3_key_prefix}/{file_name}",
-            file_name=f"/tmp/{file_name}",
-        )
+            logger.info("Uploading daily success report file to S3")
+            self.s3_service.upload_file(
+                s3_bucket_name=self.reports_bucket,
+                file_key=f"{self.s3_key_prefix}/{file_name}",
+                file_name=f"/tmp/{file_name}",
+            )
+        else:
+            logger.info("No data to report for daily success report file")
```

Comment:
Is it possible to take this out to a new function?

Reply:
It feels like scope creep to me as the only functionality that has changed properly is line 173, though I agree that the code would be cleaner this way. @abbas-khan10 what do you reckon?

Reply:
I have no preference, I had considered this when first implementing as there is repetition. What were you thinking @NogaNHS? Something like:

```
def write_and_upload_additional_reports(file_name, headers, data_rows):
    // logic here
````

Reply:
The bool would serve as a way of customising the log message for each report type as we want to be specific with which report wasn't created (open to discussion obviously)

Reply:
As opposed to doing a true/false return, I was going to suggest something more like this:
```
        if data_rows:
            logger.info("Uploading daily success report file to S3")
            self.write_and_upload_additional_reports(file_name, headers, data_rows)
        else:
            logger.info("No data to report for daily success report file")

    def write_and_upload_additional_reports(self, file_name, headers, data_rows):
        self.write_additional_report_items_to_csv(
            file_name=file_name, headers=headers, rows_to_write=data_rows
        )

        self.s3_service.upload_file(
            s3_bucket_name=self.reports_bucket,
            file_key=f"{self.s3_key_prefix}/{file_name}",
            file_name=f"/tmp/{file_name}",
        )
```

Reply:
Logic has been broken out into a new function